llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2168,8 +2185,7 @@ MemoryDepChecker::isDependent(const MemAccessInfo &A, unsigned AIdx,
   // We know that Dist is positive, but it may not be constant. Use the signed
   // minimum for computations below, as this ensures we compute the closest
   // possible dependence distance.
-  uint64_t MinDistanceNeeded =
-      TypeByteSize * *CommonStride * (MinNumIter - 1) + TypeByteSize;
```

Comment:
Changing `CommonStride` to be scaled seems like an improvement that could be split off as NFC, reducing the diff?